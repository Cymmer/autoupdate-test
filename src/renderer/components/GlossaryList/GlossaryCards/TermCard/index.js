import React from 'react';
import PropTypes from 'prop-types';
import GLOBALS from 'codechum-app-globals';
import styles from '../styles.module.scss';

import { Text, Card } from '../../../elements';
import { textTypes } from '../../../elements/constants';

const TermCard = ({ title, description }) => (
  <Card className={styles.GlossaryCard}>
    <div className={styles.GlossaryCard_title}>
      <Text
        className={styles.GlossaryCard_title_text}
        type={textTypes.HEADING.XXS}
      >
        {title}
      </Text>
    </div>
    <div className={styles.GlossaryCard_content}>
      <Text
        colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['600']}
        type={textTypes.BODY.SM}
      >
        <span dangerouslySetInnerHTML={{ __html: description }} />
      </Text>
    </div>
  </Card>
);

TermCard.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default TermCard;
