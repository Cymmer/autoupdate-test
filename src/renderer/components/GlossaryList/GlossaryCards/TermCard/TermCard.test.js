import React from 'react';
import { shallow } from 'enzyme';
import TermCard from './index';

describe('RENDERING - GlossaryList > GlossaryCards > TermCard', () => {
  it('should render with default props', () => {
    const component = shallow(
      <TermCard description="Dummy Description" title="Dummy Title" />
    );
    expect(component).toMatchSnapshot();
  });
});
