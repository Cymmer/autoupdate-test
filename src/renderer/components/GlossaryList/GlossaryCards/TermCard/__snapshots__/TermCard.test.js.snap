// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - GlossaryList > GlossaryCards > TermCard should render with default props 1`] = `
<Card
  className="GlossaryCard"
  id={null}
  onClick={null}
  onMouseEnter={null}
  onMouseLeave={null}
>
  <div
    className="GlossaryCard_title"
  >
    <Text
      className="GlossaryCard_title_text"
      colorClass="n___700"
      element={null}
      id={null}
      type="heading___xxs"
    >
      Dummy Title
    </Text>
  </div>
  <div
    className="GlossaryCard_content"
  >
    <Text
      className={null}
      colorClass="n___600"
      element={null}
      id={null}
      type="body___sm"
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "Dummy Description",
          }
        }
      />
    </Text>
  </div>
</Card>
`;
